### PR TITLE
Support converting declarations of <content> to <slot> in addition to usages.

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md has been updated

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-linter/109)

### CODE REVIEW

**Comment by rictic** [comment]
Hm, on reflection, we probably shouldn't be doing this automatically... probably.

Need to think about this a bit more I think.

**Comment by rictic** [comment]
Updated to not apply this automatically. Should follow up and give a better way to warn without automatically fixing.

File: src/polymer/content-to-slot-declarations.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        const warning = new FixableWarning({
+          code: 'content-to-slot-declaration',
+          message:
+              `<content> tags are part of the deprecated Shadow Dom v0 API. ` +
+              `Replace with a <slot> tag.`,
+          parsedDocument,
+          severity: Severity.WARNING,
+          sourceRange: parsedDocument.sourceRangeForStartTag(contentNode)!
+        });
+        const attrs = [...contentNode.attrs];
```
Comment (valdrinkoshi) [inline src/polymer/content-to-slot-declarations.ts]: got lost on what's happening in this block (here to line 85). Comments/split it in a separate function maybe?

File: test/content-to-slot-declarations/before-fixes.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<dom-module id="foo-bar">
+  <template>
+    <content></content>
```
Comment (valdrinkoshi) [inline test/content-to-slot-declarations/before-fixes.html]: nit: move as last `<content>`

File: test/content-to-slot-declarations/after-fixes.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<dom-module id="foo-bar">
+  <template>
+    <slot></slot>
```
Comment (valdrinkoshi) [inline test/content-to-slot-declarations/after-fixes.html]: nit: move as last `<slot>`

File: src/polymer/content-to-slot-declarations.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        const selectorAttr = attrs.find((a) => a.name === 'select');
+        const selector = selectorAttr && selectorAttr.value;
+        let slotName = null;
+        if (selector) {
+          slotName = slotNameForSelector(selector);
+          while (slots.has(slotName)) {
+            slotName += '-unique-suffix';
```
Comment (valdrinkoshi) [inline src/polymer/content-to-slot-declarations.ts]: could you add a test for this? e.g.
Before:
```html
<content select="#foo">
  Foo
</content>
<content select=".foo">
  Foo
</content>
```
After:
```html
<slot name="foo" old-content-selector="#foo">
  Foo
</slot>
<slot name="foo-unique-suffix" old-content-selector=".foo">
  Foo
</slot>

File: src/polymer/content-to-slot-declarations.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+          parsedDocument,
+          severity: Severity.WARNING,
+          sourceRange: parsedDocument.sourceRangeForStartTag(contentNode)!
+        });
+        const attrs = [...contentNode.attrs];
+        const selectorAttr = attrs.find((a) => a.name === 'select');
+        const selector = selectorAttr && selectorAttr.value;
+        let slotName = null;
+        if (selector) {
+          slotName = slotNameForSelector(selector);
```
Comment (valdrinkoshi) [inline src/polymer/content-to-slot-declarations.ts]: Selectors like ".baz .foo" should not be fixable, e.g. 
```html
<content select=".baz .foo">
```
currently gets updated to:
```html
<slot name="baz" old-content-selector=".baz .foo">
```

File: src/polymer/content-to-slot-declarations.ts (hunk 1 above)
Comment (rictic) [reply src/polymer/content-to-slot-declarations.ts]: +1, done

File: src/polymer/content-to-slot-declarations.ts (hunk 4 above)
Comment (rictic) [reply src/polymer/content-to-slot-declarations.ts]: Done

File: test/content-to-slot-declarations/after-fixes.html (hunk 3 above)
Comment (rictic) [reply test/content-to-slot-declarations/after-fixes.html]: Done

File: test/content-to-slot-declarations/before-fixes.html (hunk 2 above)
Comment (rictic) [reply test/content-to-slot-declarations/before-fixes.html]: Done

File: src/polymer/content-to-slot-declarations.ts (hunk 5 above)
Comment (rictic) [reply src/polymer/content-to-slot-declarations.ts]: Hm, yeah good point! 

I agree, but I think that's something to add to the usages PR, as that sort of `<content>` element still needs to be upgraded to a `<slot>`, and I think it makes sense to preserve the old content selector as it is.

I'll add a check to the usages PR, I wanted to do a better job with selectors anyways.

**Review by valdrinkoshi — APPROVED** [review]
LGTM 👍

**Pull request merged**